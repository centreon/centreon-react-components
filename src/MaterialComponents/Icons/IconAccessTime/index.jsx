import React from 'react';
import AccessTimeIcon from '@material-ui/icons/AccessTime';
import MaterialIcon from '../MaterialIcon';

function IconAccessTime(props) {
  return (
    <MaterialIcon {...props}>
      <AccessTimeIcon />
    </MaterialIcon>
  );
}

export default IconAccessTime;
