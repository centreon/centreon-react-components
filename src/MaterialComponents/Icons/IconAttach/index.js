/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */

import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Attach from '@material-ui/icons/AttachFile';

const useStyles = makeStyles(() => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  icon: {
    color: '#707070',
    cursor: 'pointer',
    fontSize: 14,
    verticalAlign: 'middle',
  },
  iconAttach: {
    width: 80,
    height: 49,
    textAlign: 'center',
    lineHeight: '49px',
    backgroundColor: '#fff',
    display: 'inline-block',
    marginRight: 15,
    verticalAlign: 'middle',
    cursor: 'pointer',
    marginLeft: -20,
  },
  iconAttachLabel: {
    fontSize: 12,
    color: '#707070',
    display: 'inline-block',
    verticalAlign: 'middle',
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    marginLeft: 3,
  },
}));

function IconAttach({ customStyle, onClick }) {
  const classes = useStyles();

  return (
    <React.Fragment>
      <span onClick={onClick} className={classes.iconAttach}>
        <Attach style={customStyle} className={classes.icon} />
        <span className={classes.iconAttachLabel}>ICON</span>
      </span>
    </React.Fragment>
  );
}

export default IconAttach;
