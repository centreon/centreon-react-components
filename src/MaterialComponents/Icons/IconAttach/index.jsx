import React from 'react';
import Attach from '@material-ui/icons/AttachFile';
import styled from '@emotion/styled';
import MaterialIcon from '../MaterialIcon';

const AttachMaterial = styled(MaterialIcon)(() => ({
  width: 80,
  height: 49,
  textAlign: 'center',
  lineHeight: '49px',
  marginRight: 15,
  marginLeft: -20,
}));

const MiniAttach = styled(Attach)(() => ({
  fontSize: 14,
  verticalAlign: 'middle',
}));

const Label = styled.span(() => ({
  fontSize: 12,
  fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
  marginLeft: 3,
}));

function IconAttach(props) {
  return (
    <AttachMaterial {...props}>
      <MiniAttach />
      <Label>ICON</Label>
    </AttachMaterial>
  );
}

export default IconAttach;
