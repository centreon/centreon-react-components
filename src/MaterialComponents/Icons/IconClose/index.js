/* eslint-disable react/jsx-filename-extension */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/prop-types */

import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Close from '@material-ui/icons/Close';

const useStyles = makeStyles(() => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  icon: {
    color: '#424242',
    cursor: 'pointer',
    float: 'right',
    fontSize: 32,
    zIndex: 9,
    position: 'absolute',
    right: 11,
    top: 8,
  },
  iconLabel: {
    color: '#009fdf',
    fontSize: 12,
    display: 'inline-block',
    verticalAlign: 'middle',
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 'bold',
    cursor: 'pointer',
    paddingLeft: 5,
  },
  iconCustomStyle: {
    content: '""',
    position: 'absolute',
    right: 0,
    top: 0,
    width: 54,
    height: 49,
    cursor: 'pointer',
    '&:hover': {
      background: '#707070',
      '& svg': {
        color: '#fff',
      },
    },
  },
}));

function IconClose({ label, customStyle, onClick }) {
  const classes = useStyles();

  return (
    <span onClick={onClick} className={classes.iconCustomStyle}>
      <Close style={customStyle} className={classes.icon} />
      {label && <span className={classes.iconLabel}>{label}</span>}
    </span>
  );
}

export default IconClose;
