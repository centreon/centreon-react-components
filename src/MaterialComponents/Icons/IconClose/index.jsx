/* eslint-disable react/prop-types */

import React from 'react';
import Close from '@material-ui/icons/Close';
import styled from '@emotion/styled';
import MaterialIcon from '../MaterialIcon';

const Label = styled.span(() => ({
  color: '#009fdf',
  fontSize: 12,
  display: 'inline-block',
  verticalAlign: 'middle',
  fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
  fontWeight: 'bold',
  cursor: 'pointer',
  paddingLeft: 5,
}));

const FloatingIcon = styled(MaterialIcon)(() => ({
  content: '""',
  position: 'absolute',
  right: 0,
  top: 0,
  width: 54,
  height: 49,
  '&:hover': {
    background: '#707070',
    '& svg': {
      color: '#fff',
    },
  },
}));

const GreyClose = styled(Close)(() => ({
  color: '#424242',
  cursor: 'pointer',
  float: 'right',
  fontSize: 32,
  zIndex: 9,
  position: 'absolute',
  right: 11,
  top: 8,
}));

function IconClose({ label, ...rest }) {
  return (
    <FloatingIcon {...rest}>
      <GreyClose />
      {label && <Label>{label}</Label>}
    </FloatingIcon>
  );
}

export default IconClose;
