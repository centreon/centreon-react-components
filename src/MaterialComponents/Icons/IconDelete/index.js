import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Delete from "@material-ui/icons/Delete";

const useStyles = makeStyles(() => ({
  root: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center"
  },
  icon: {
    color: "#707070",
    cursor: "pointer"
  },
  iconWrap: {
    display: "inline-block",
    verticalAlign: "middle",
    height: 24
  }
}));

function IconDelete({ customStyle, ...rest }) {
  const classes = useStyles();

  return (
    <span {...rest} className={classes.iconWrap}>
      <Delete style={customStyle} className={classes.icon} />
    </span>
  );
}

export default IconDelete;
