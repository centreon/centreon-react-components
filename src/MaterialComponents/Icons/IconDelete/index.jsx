import React from 'react';
import Delete from '@material-ui/icons/Delete';
import MaterialIcon from '../MaterialIcon';

function IconDelete(props) {
  return (
    <MaterialIcon {...props}>
      <Delete />
    </MaterialIcon>
  );
}

export default IconDelete;
