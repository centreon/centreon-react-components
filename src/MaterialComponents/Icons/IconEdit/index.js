import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Edit from '@material-ui/icons/Edit';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  icon: {
    color: '#0072CE',
    cursor: 'pointer',
    fontSize: 21,
    position: 'absolute',
    right: 3,
  },
}));

function IconEdit({customStyle, onClick}) {
  const classes = useStyles();

  return (
    <Edit onClick={onClick} style={customStyle} className={classes.icon} />
  );
}

export default IconEdit;
