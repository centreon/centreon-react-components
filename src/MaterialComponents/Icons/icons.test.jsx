/* eslint-disable no-undef */

import React from 'react';
import { render } from '@testing-library/react';
import IconAccessTime from './IconAccessTime';
import IconAttach from './IconAttach';
import IconClose from './IconClose';
import IconDone from './IconDone';
import IconEdit from './IconEdit';
import IconDelete from './IconDelete';
import IconInsertChart from './IconInsertChart';
import IconLibraryAdd from './IconLibraryAdd';
import IconPowerSettings from './IconPowerSettings';
import IconPowerSettingsDisable from './IconPowerSettingsDisable';
import IconRefresh from './IconRefresh';
import IconReportProblem from './IconReportProblem';
import IconVisible from './IconVisible';
import IconInvisible from './IconInvisible';

[
  IconAccessTime,
  IconAttach,
  IconClose,
  IconDone,
  IconEdit,
  IconDelete,
  IconInsertChart,
  IconLibraryAdd,
  IconPowerSettings,
  IconPowerSettingsDisable,
  IconRefresh,
  IconReportProblem,
  IconVisible,
  IconInvisible,
].forEach((IconComponent) => {
  describe(IconComponent, () => {
    it('renders', () => {
      const { container } = render(<IconComponent />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
