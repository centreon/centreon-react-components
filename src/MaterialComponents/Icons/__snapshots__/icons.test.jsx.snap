// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconAccessTime renders 1`] = `
<span
  class="span-root-1 span-root-2"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
    <path
      d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
    />
  </svg>
</span>
`;

exports[`IconAttach renders 1`] = `
<span
  class="makeStyles-iconAttach-14"
/>
`;

exports[`IconClose renders 1`] = `
<span
  aria-label="icon close"
  class="span-root-19 span-root-20 MaterialIcon-root-17 MaterialIcon-root-18"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root CloseIcon-root-21 CloseIcon-root-22"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
    />
  </svg>
</span>
`;

exports[`IconDelete renders 1`] = `
<span
  class="span-root-54 span-root-55"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
    />
  </svg>
</span>
`;

exports[`IconDone renders 1`] = `
<span
  class="span-root-32 span-root-33"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
    />
  </svg>
</span>
`;

exports[`IconEdit renders 1`] = `
.emotion-0 {
  color: #0072CE;
  font-size: 21px;
  position: absolute;
  right: 3px;
}

<span
  class="span-root-43 span-root-44"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root emotion-0"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
    />
  </svg>
</span>
`;

exports[`IconInsertChart renders 1`] = `
<span
  class="span-root-65 span-root-66"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z"
    />
  </svg>
</span>
`;

exports[`IconInvisible renders 1`] = `
<span
  class="span-root-142 span-root-143"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="m0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
    />
  </svg>
</span>
`;

exports[`IconLibraryAdd renders 1`] = `
<span
  class="span-root-76 span-root-77"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9h-4v4h-2v-4H9V9h4V5h2v4h4v2z"
    />
  </svg>
</span>
`;

exports[`IconPowerSettings renders 1`] = `
.emotion-0 {
  color: #fff;
  background-color: #707070;
  border-radius: 50%;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  padding: 4px;
}

<span
  aria-label="icon enable disable"
  class="span-root-87 span-root-88"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root emotion-0"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M13 3h-2v10h2V3zm4.83 2.17l-1.42 1.42C17.99 7.86 19 9.81 19 12c0 3.87-3.13 7-7 7s-7-3.13-7-7c0-2.19 1.01-4.14 2.58-5.42L6.17 5.17C4.23 6.82 3 9.26 3 12c0 4.97 4.03 9 9 9s9-4.03 9-9c0-2.74-1.23-5.18-3.17-6.83z"
    />
  </svg>
</span>
`;

exports[`IconPowerSettings renders 2`] = `
.emotion-0 {
  color: #fff;
  background-color: #707070;
  border-radius: 50%;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  padding: 4px;
}

.emotion-1 {
  display: inline-block;
  vertical-align: middle;
  height: 23px;
  width: 23px;
  position: relative;
}

.emotion-1::after {
  content: '';
  position: absolute;
  width: 2px;
  height: 30px;
  background: #7f7f7f;
  -webkit-transform: rotate(140deg);
  -ms-transform: rotate(140deg);
  transform: rotate(140deg);
  left: 9px;
  top: -2px;
  z-index: 1;
}

.emotion-1::before {
  content: '';
  position: absolute;
  width: 4px;
  height: 30px;
  background: #fff;
  -webkit-transform: rotate(140deg);
  -ms-transform: rotate(140deg);
  transform: rotate(140deg);
  left: 9px;
  top: -3px;
  z-index: 1;
}

<span
  aria-label="icon enable disable inactive"
  class="span-root-98 span-root-99 emotion-1"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root emotion-0"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M13 3h-2v10h2V3zm4.83 2.17l-1.42 1.42C17.99 7.86 19 9.81 19 12c0 3.87-3.13 7-7 7s-7-3.13-7-7c0-2.19 1.01-4.14 2.58-5.42L6.17 5.17C4.23 6.82 3 9.26 3 12c0 4.97 4.03 9 9 9s9-4.03 9-9c0-2.74-1.23-5.18-3.17-6.83z"
    />
  </svg>
</span>
`;

exports[`IconRefresh renders 1`] = `
<span
  class="span-root-109 span-root-110"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
    />
  </svg>
</span>
`;

exports[`IconReportProblem renders 1`] = `
<span
  class="span-root-120 span-root-121"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
    />
  </svg>
</span>
`;

exports[`IconVisible renders 1`] = `
<span
  class="span-root-131 span-root-132"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
    />
  </svg>
</span>
`;
